### PR TITLE
Pull from dev into master for end of sprint 1

### PR DESCRIPTION
**Pull request by cameronwolf**
Create Error Message Functionality #35 
Validate GET request for outfitters permit #57
Validate GET request for noncommercial permit #56 
Add instructions to README.md #75 
Add fallback for env var PORT #76 
Validate POST request for outfitters permit #55 
Validate POST request for noncommercial permit #54 
Add 200 success message to Swagger docs #79

### CODE REVIEW

File: controllers/permits/special-uses/validate.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+*/
+
+//*******************************************************************
+
+'use strict';
+
+//*******************************************************************
+
+/*
+var permit_id = function(id){
```
Comment (brianfunk) [inline controllers/permits/special-uses/validate.js]: need to remove comments code if unused

File: controllers/permits/special-uses/validate.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    }
+    
+    return valid;
+
+};
+*/
+
+var applicant_info = function(req){
+
+	var output = {
```
Comment (brianfunk) [inline controllers/permits/special-uses/validate.js]: need consistent indentation

File: controllers/permits/special-uses/non-commercial/validate.js [hunk 3]
```diff
@@ -13,21 +13,38 @@
...
+var noncommercial = function(req){
+
+	var output = {
+		'fields_valid': true,
+		'error_array':[]
+	};
+
+	if (!req.body['noncommercial-fields'].activityDescription) {
+        output.fields_valid = false;
+        output.error_array.push('activityDescription');
```
Comment (lpendem) [inline controllers/permits/special-uses/non-commercial/validate.js]: a function can be used for these two lines of code.

File: controllers/permits/special-uses/validate.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        output.error_array.push('firstName');
+    }
+    if(!req.body['applicant-info'].lastName){
+        output.fields_valid = false;
+        output.error_array.push('lastName');
+    }
+    if(!req.body['applicant-info'].dayPhone){
+        output.fields_valid = false;
+        output.object_missing_message = 'dayPhone cannot be empty.';
+    }else{
```
Comment (brianfunk) [inline controllers/permits/special-uses/validate.js]: need consistent formation - eg separate lines for if/else

File: docs/swagger.json [hunk 5]
```diff
@@ -106,11 +112,14 @@
                   "description": "Permit object that needs to be updated",
                   "required": true,
                   "schema": {
-                     "$ref": "#/definitions/non-commercial-permit"
+                     "$ref": "#/definitions/noncommercial-permit"
                   }
                }
             ],
             "responses": {
+               "200": {
```
Comment (brianfunk) [inline docs/swagger.json]: added 200 - good!

File: index.js [hunk 6]
```diff
@@ -25,12 +25,14 @@ var fsr = require('file-stream-rotator');
...
 
+var body_parser = require('body-parser');
+
 var routes = require('./routes');
 
 //*******************************************************************
 // environment variables
 
-var PORT = process.env.PORT;
+var PORT = process.env.PORT || 8000;
```
Comment (brianfunk) [inline index.js]: added default port - good!

File: routes/permits/special-uses/non-commercial/index.js [hunk 7]
```diff
@@ -19,6 +19,7 @@ var router = express.Router();
 var include = require('include')(__dirname);
 
 var non_commercial = include('controllers/permits/special-uses/non-commercial');
```
Comment (brianfunk) [inline routes/permits/special-uses/non-commercial/index.js]: need to change file name to noncommercial (remove -) for consistency with naming convention

File: test/validate-noncommercial-fields-test.js [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+    });
+
+    it('should return valid json with a 400 status code for noncommercial POST request without a endDateTime', function(done) {
+        request(server)
+            .post('/permits/special-uses/noncommercial')
+            .send(
+            	util.update_input_data(
+            		post_input,
+            		{
+						"noncommercial-fields": {
```
Comment (brianfunk) [inline test/validate-noncommercial-fields-test.js]: need consistent indentation and formating

**Pull request merged**